pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -695,3 +695,13 @@ def test_loc_getitem_index_differently_ordered_slice_none():
         columns=["a", "b"],
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_loc_getitem_drops_levels_for_one_row_dataframe():
+    # GH#10521
+    df = DataFrame({"a": ["a"], "b": ["b"], "c": ["a"], "d": 0}).set_index(
```

Comment:
personal preference: to make it super-obvious what is and isn't being tested -- in particular that `set_index` is not being tested -- id prefer this as

```
mi = MultiIndex.from_product(...)
df = DataFrame({"d": 0}, index=mi)
```

Reply:
this would also be easier to follow if the values and the columns didnt overlap

Reply:
Thx, simplified it